src/Mvc/src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationFactory.cs

```diff
@@ -119,6 +123,21 @@ private void EnsureServer()
 
             EnsureDepsFile();
 
+            var hostBuilder = HostFactoryResolver.ResolveHostBuilderFactory<IHostBuilder>(typeof(TEntryPoint).Assembly)?.Invoke(Array.Empty<string>());
```

Comment:
If this is going to be the default option in the template, we should have a protected virtual CreateHostBuilder implementation on this class that does the same thing as CreateWebHostBuilder or that falls back to CreateWebHostBuilder when it can't resolve the host so that we can produce meaningful errors in this case.

Whenever we make a convention, we need an extensibility point as users are free to not follow the pattern and the class should still be usable.